vtStor/DriveManager.cpp

```diff
@@ -30,6 +34,15 @@ cDriveManager::~cDriveManager()
     m_Drives.clear();
 }
 
+void cDriveManager::GetInstance(std::unique_ptr<cDriveManagerInterface>& DriveManager)
+{
+    if (nullptr == s_Instance)
+    {
+        s_Instance = std::unique_ptr<cDriveManager>(new cDriveManager());
+    }
+    DriveManager = std::move(s_Instance);
```

Comment:
Reference from s_Instance would be lost? Why not just use shared_ptr? It's still a singleton with shared_ptr.